app/codecs/CirceDecoders.scala

```diff
@@ -58,9 +62,24 @@ object CirceDecoders {
     Decoder.decodeString.emap(code => BillingPeriod.fromString(code).toRight(s"Unrecognised period code '$code'"))
   implicit val encodePeriod: Encoder[BillingPeriod] = Encoder.encodeString.contramap[BillingPeriod](_.toString)
 
+  implicit val ophanIdsCodec: Codec[OphanIds] = deriveCodec
```

Comment:
You should be able to import most of these implicits from acquisitions event producer.

Reply:
Ok, as discussed, run as is and refactor in a separate pull request.